blob/repro_test.go

```diff
@@ -0,0 +1,64 @@
+package blob_test
+
+import (
+	"testing"
+
+	"github.com/celestiaorg/celestia-app/v2/pkg/proof"
+	"github.com/celestiaorg/celestia-node/blob"
+	"github.com/celestiaorg/nmt"
+	"github.com/celestiaorg/nmt/pb"
+)
+
+// Reported at https://github.com/celestiaorg/celestia-node/issues/3731.
+func TestCommitmentProofRowProofVerifyWithEmptyRoot(t *testing.T) {
```

Comment:
Why not aggregate all this test as sub-tests? Example:

```
func TestCommitmentProofHardening(...) {
     t.Run("verify with empty root", func() ... {
     })

     t.Run("verify row proof", func() ... {
     })
}
```

WDYT ?

Reply:
I highly recommend keeping regression tests separate and isolated. Perhaps after this merge as the maintainers y'all can make the changes as you desire? But in my experience of dealing with a large amount of regressions and security+QA for large projections, best to always keep the tests as isolated as possible.

Reply:
Disagree on "separate and isolated" but "can make the changes as you desire" sounds fair.